### PR TITLE
Tidy the project

### PR DESCRIPTION
**Pull request by suhr**

### CODE REVIEW

File: sana_derive/src/lib.rs [hunk 1]
```diff
@@ -130,6 +131,34 @@ fn build_spec(source: ItemEnum) -> SanaSpec {
...
+///
+/// - `#[error]`: Marks the given variant as the error variant. There must be
+/// exactly one error variant for a given enum
+/// - `#[regex(re)]`: specify the regular expression corresponding to
+/// the given variant
+/// - `#[token(tok)]`: specify the string corresponding to the given variant
+///
+/// Attributes `regex` and `token` can also receive the following parameters:
+/// 
+/// - `priority = <integer>`: the priority for the rule
```
Comment (kpp) [inline sana_derive/src/lib.rs:146]: default priority?

File: sana/src/lib.rs [hunk 2]
```diff
@@ -140,12 +140,12 @@ impl<'input, Token: Sana> Lexer<'input, Token> {
     }
 }
 
-/// Token together with its range
+/// A value (for example, token) together with its range
 #[derive(Debug, Clone, Copy, PartialEq)]
 pub struct Spanned<T> {
     pub start: usize,
     pub end: usize,
```
Comment (kpp) [inline sana/src/lib.rs:147]: including/excluding?

File: sana_derive/src/lib.rs [hunk 3]
```diff
@@ -143,11 +141,11 @@ fn build_spec(source: ItemEnum) -> SanaSpec {
 ///
 /// Attributes `regex` and `token` can also receive the following parameters:
 /// 
-/// - `priority = <integer>`: the priority for the rule
+/// - `priority = <integer>` (default is `0`): the priority for the rule
 ///
 /// # Regular expression syntax
 ///
-/// Regular expression, passed to attribute `regex`, has the following syntax:
+/// Regular expression, passed to attribute `regex`, have the following syntax:
```
Comment (kpp) [inline sana_derive/src/lib.rs:148]:
```suggestion
/// Regular expressions, passed to attribute `regex`, have the following syntax:
```

or

```suggestion
/// Regular expression, passed to attribute `regex`, has the following syntax:
```

File: README.md [hunk 4]
```diff
@@ -1,4 +1,81 @@
...
 
-Sana is a lexer generator for Rust.
+Sana is a lexer generator for Rust. It provides a easy way to create a lexical analyzer for a language.
+
+Unlike other lexer generator, Sana supports extended regular expressions, which allow you to express your intent more clearly. For example, you can write `"[[:punct:]]+" & !".*--.*"` to represent a sequence of punctuation characters which does not contain `--`.
+
+At compile time, Sana:
+
+- Constructs an deterministic state automata from the token definitions
+- Generates code from the automata
```
Comment (kpp) [inline README.md:11]:
```suggestion
- TODO: Generates code from the automata
```

Comment (suhr) [reply README.md:11]: Not quite a TODO, since IR is also a kind of code. But fair enough, I should make it more clear.

File: README.md [hunk 5]
```diff
@@ -1,4 +1,81 @@
 # Sana
 [![Rust](https://github.com/suhr/sana/workflows/Rust/badge.svg)](https://github.com/suhr/sana/actions?query=branch%3Amaster)
 
-Sana is a lexer generator for Rust.
+Sana is a lexer generator for Rust. It provides a easy way to create a lexical analyzer for a language.
```
Comment (kpp) [inline README.md:4]:
```suggestion
Sana is a lexer generator for Rust. It provides a easy way to create a lexical analyzer and can be integrated with lalrpop.
```

Comment (suhr) [reply README.md:4]: `lalrpop` is not that special.

Comment (kpp) [reply README.md:4]: ... and other parser generators.

**Pull request merged**